binding.c

```diff
@@ -234,6 +239,98 @@ bare_os_hostname(js_env_t *env, js_callback_info_t *info) {
   return result;
 }
 
+static js_value_t *
+bare_os_userinfo (js_env_t *env, js_callback_info_t *info) {
+  int err;
+  js_value_t *result;
+  err = js_create_object(env, &result);
+  if (err < 0) return NULL;
+
+  #ifdef _WIN32
+  uv_passwd_t pwd;
+  err = uv_os_get_passwd(&pwd);
+  if (err != 0) {
+      js_throw_error(env, "Error", "Could not retrieve user info");
+      return NULL;
+  }
+
+  js_value_t *uid_value;
+  err = js_create_int32(env, (int32_t)-1, &uid_value);
+  if (err < 0) return NULL;
+  err = js_set_named_property(env, result, "uid", uid_value);
+  if (err < 0) return NULL;
+
+  js_value_t *gid_value;
+  err = js_create_int32(env, (int32_t)-1, &gid_value);
+  if (err < 0) return NULL;
+  err = js_set_named_property(env, result, "gid", gid_value);
+  if (err < 0) return NULL;
+
+  js_value_t *username_value;
+  err = js_create_string_utf8(env, (utf8_t *) pwd.username, strlen(pwd.username), &username_value);
+  if (err < 0) return NULL;
+  err = js_set_named_property(env, result, "username", username_value);
+  if (err < 0) return NULL;
+
+  js_value_t *home_dir_value;
+  err = js_create_string_utf8(env, (utf8_t *) pwd.homedir, strlen(pwd.homedir), &home_dir_value);
+  if (err < 0) return NULL;
+  err = js_set_named_property(env, result, "homedir", home_dir_value);
+  if (err < 0) return NULL;
+
+  js_value_t *shell_value;
+  err = js_get_null(env, &shell_value);
+  if (err < 0) return NULL;
+  err = js_set_named_property(env, result, "shell", shell_value);
+  if (err < 0) return NULL;
+
+  uv_os_free_passwd(&pwd);
+  return result;
```

Comment:
This should be the _only_ implementation, there's no need for the ad-hoc reimplementation of `uv_os_get_passwd()` further down.